test/data/test_modules.py

```diff
@@ -213,11 +190,94 @@ def test_broadcast_scaled_dot_product(self):
         )
         self.assertEqual(sdp_attn_output, sdp_attn_output_full)
         self.assertEqual(sdp_attn_weights, sdp_attn_weights_full)
-        # attn_mask's dim -3 is not equal to neither batch size or 1
-        with self.assertRaisesRegex(RuntimeError, "The size of the attn_mask is not correct."):
-            SDP(
-                query.expand(tgt_len, bsz * nhead, embed_dim),
-                key.expand(src_len, bsz * nhead, embed_dim),
-                value.expand(src_len, bsz * nhead, embed_dim),
-                attn_mask=attn_mask_2D.expand(2, tgt_len, src_len),
-            )
+
+    def test_assert_raises(self):
```

Comment:
This test name is generic and thus it is making it hard to understand what test is about.

Also can you add docstring so to tell the intended behavior of the test? Typically this should be "(Under some condition) this feature should produce this result"

Reply:
Done. See #1667.